server/conn.go

```diff
@@ -365,11 +421,21 @@ func (cc *clientConn) readOptionalSSLRequestAndHandshakeResponse() error {
 		return errors.Trace(err)
 	}
 
+	isOldVersion := false
+
 	var resp handshakeResponse41
 
 	pos, err := parseHandshakeResponseHeader(&resp, data)
 	if err != nil {
-		return errors.Trace(err)
+		if err == mysql.ErrMalformPacket {
```

Comment:
>Protocol::HandshakeResponse41:
Handshake Response Packet sent by 4.1+ clients supporting CLIENT_PROTOCOL_41 capability, if the server announced it in its Initial Handshake Packet. Otherwise (talking to an old server) the Protocol::HandshakeResponse320 packet must be used.

https://dev.mysql.com/doc/internals/en/connection-phase-packets.html

@imtbkcat  can we direct use `CLIENT_PROTOCOL_41 capability` to decide use 41 or 32, because our Initial Handshake Packet always contain `ClientProtocol41`?

parse handshake twice has risk